File: python/strategy/ActionGame/behavior/WeaponBehavior.py

class WeaponBehavior:
    def use_weapon(self):
        return

class Axe(WeaponBehavior):

    def __init__(self, character_name):
        self.__character_name = character_name

    def use_weapon(self):
        print("the " + self.__character_name + " is using Axe")

class Sword(WeaponBehavior):

    def __init__(self, character_name):
        self.__character_name = character_name

    def use_weapon(self):
        print("the " + self.__character_name + " is using Sword")

class BowAndArrow(WeaponBehavior):

    def __init__(self, character_name):
        self.__character_name = character_name

    def use_weapon(self):
        print("the " + self.__character_name + " is using BowAndArrow")

class Knife(WeaponBehavior):

    def __init__(self, character_name):
        self.__character_name = character_name

    def use_weapon(self):
        print("the " + self.__character_name + " is using Knife")

File: C++/Strategy/ActionGame/behavior/WeaponBehavior.cpp
#include <string>
#include <iostream>
using namespace std;

class WeaponBehavior {

public:
    virtual void useWeapon() = 0;
};

class Axe : public WeaponBehavior {

public:
    Axe(string characterName) : characterName(characterName) {}
    void useWeapon() override {
        cout << "the " << characterName << " is using Axe" << endl;
    }

private:
    string characterName;
};

class Sword : public WeaponBehavior {

public:
    Sword(string characterName) : characterName(characterName) {}
    void useWeapon() override {
        cout << "the " << characterName << " is using Sword" << endl;
    }

private:
    string characterName;
};

class BowAndArrow : public WeaponBehavior {

public:
    BowAndArrow(string characterName) : characterName(characterName) {}
    void useWeapon() override {
        cout << "the " << characterName << " is using BowAndArrow" << endl;
    }

private:
    string characterName;
};

class Knife : public WeaponBehavior {

public:
    Knife(string characterName) : characterName(characterName) {}
    void useWeapon() override {
        cout << "the " << characterName << " is using Knife" << endl;
    }

private:
    string characterName;
};


